lib/modules/manager/maven/extract.ts

```diff
@@ -505,7 +510,7 @@ export async function extractAllPackageFiles(
       for (const dep of pkgFile.deps) {
         /* istanbul ignore else */
         if (dep.registryUrls) {
-          dep.registryUrls.push(...additionalRegistryUrls);
+          dep.registryUrls.unshift(...additionalRegistryUrls);
```

Comment:
This will invert the order of registryUrls provided by the user.

Reply:
It should prepend the urls from the settings file (additionalRegistryUrls). They should go before the registryUrls from the pom file.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift#description